### ISSUE TITLE
Interface Segregation Principle design flaw

### ISSUE THREAD

**Issue by namagi** [jupeter/clean-code-php#147]
In [ISP](https://github.com/jupeter/clean-code-php/blob/master/README.md#interface-segregation-principle-isp) there are severe design flaws which clearly indicate you have no experience with goverment employees:
```php
class Human implements Employee
{
    public function work(): void
    {
        // ....working  --> instance would likely throw UnknownOperationException
    }

    public function eat(): void
    {
        // ...... eating in lunch break --> entire working hours IS-A lunch break; it may or may not have anything to do with eating
    }
}
```

Also, in **Good:** _Not every worker is an employee, but every employee is a worker_. --> Again, you have not met goverment employees.

All this makes it confusing to grasp the concept for some instances   :wink:

**Comment by peter-gribanov** [comment]
You always can give another example in PR

**Comment by spacemudd** [comment]
Surely this baseless issue is not serious, right? 😅

The example about ISP is an example of ISP. Not an example about working with private/govt. employees. It's succinctly explained without building an authentic, ready-for-production interface, haha.

>which clearly indicate you have no experience with goverment employees:

I really believe this issue is a joke? haha

**Comment by archfrog** [comment]
Was laughing ... a bit :-)  Besides, shouldn't it be `class Employee implements Human`?

But yes, I think somebody had success in being funny.

**Comment by peter-gribanov** [comment]
> Besides, shouldn't it be `class Employee implements Human`?

@archfrog no, because it is not a human in global context, but a human in context of employees.
